src/main/java/org/folio/consortia/service/impl/TenantServiceImpl.java

```diff
@@ -149,4 +170,58 @@ private ConsortiaConfiguration createConsortiaConfigurationBody(String tenantId)
     configuration.setCentralTenantId(tenantId);
     return configuration;
   }
+
+  private void createShadowAdminUserWithPermissions(UUID userId) {
+    User userOptional = userTenantService.getUser(userId);
+    if (Objects.isNull(userOptional.getId())) {
+      userOptional = createUser(userId);
+    }
+    Optional<PermissionUser> permissionUserOptional = permissionsClient.get("userId==" + userOptional.getId())
+      .getPermissionUsers()
+      .stream()
+      .findFirst();
+    if (permissionUserOptional.isEmpty()) {
+      createPermissionUser(userOptional.getId());
+    }
+  }
+
+  private PermissionUser createPermissionUser(String userId) {
+    List<String> perms = readPermissionsFromResource(PERMISSIONS_FILE_PATH);
+
+    if (CollectionUtils.isEmpty(perms)) {
+      throw new IllegalStateException("No user permissions found in " + PERMISSIONS_FILE_PATH);
+    }
+
+    var permissionUser = PermissionUser.of(UUID.randomUUID().toString(), userId, perms);
+    log.info("Creating {}.", permissionUser);
+    return permissionsClient.create(permissionUser);
+  }
+
+  private List<String> readPermissionsFromResource(String permissionsFilePath) {
+    List<String> result = new ArrayList<>();
+    var url = Resources.getResource(permissionsFilePath);
+
+    try {
+      result = Resources.readLines(url, StandardCharsets.UTF_8);
+    } catch (IOException e) {
+      log.error(String.format("Can't read user permissions from %s.", permissionsFilePath), e);
+    }
+
+    return result;
+  }
+
+  private User createUser(UUID userId) {
+    var result = createUserObject(userId);
+    log.info("Creating {}.", result);
+    usersClient.saveUser(result);
+    return result;
+  }
+
+  private User createUserObject(UUID userId) {
+    final var result = new User();
+    result.setId(userId.toString());
+    result.setUsername("Admin");
```

Comment:
user name is incorrect, in fact you need to create real shadow user

Reply:
fixed